eve/main.yml

```diff
@@ -405,6 +405,14 @@ stages:
               major=$(echo "%(prop:product_version)s" | cut -d'.' -f1) &&
               minor=$(echo "%(prop:product_version)s" | cut -d'.' -f2) &&
               echo "$major.$(( $minor-1 ))"
+      - SetPropertyFromCommand:
+          name: Set previous patch version to upgrade from and downgrade to
+          property: product_patch_version_prev
+          command: >
+              major=$(echo "%(prop:product_version)s" | cut -d'.' -f1) &&
+              minor=$(echo "%(prop:product_version)s" | cut -d'.' -f2) &&
+              patch=$(echo "%(prop:product_version)s" | cut -d'.' -f3) &&
+              echo "$major.$minor.$(( $patch-1 ))"
```

Comment:
You must do some checks before doing that, otherwise you could end up with a version `2.5.-1`, not sure that what we want. :)
I would do something like:
```
              major=$(echo "%(prop:product_version)s" | cut -d'.' -f1) &&
              minor=$(echo "%(prop:product_version)s" | cut -d'.' -f2) &&
              patch=$(echo "%(prop:product_version)s" | cut -d'.' -f3) &&
              echo "$major.$minor.$(( patch > 0 ? patch - 1 : 0 ))"
```

Reply:
As agreed in person, we will deactivate upgrade testing for patch versions on the 2.5 branch since `doStepIf` is buggy at this time.

Reply:
> Would be very nice to have the `doStepIf` for such cases.

Working on it :)